lldb/source/Target/VerboseTrapFrameRecognizer.cpp

```diff
@@ -36,10 +37,23 @@ static StackFrameSP FindMostRelevantFrame(Thread &selected_thread) {
     if (!frame_name)
       return nullptr;
 
-    // Found a frame outside of the `std` namespace. That's the
+    // Find a frame outside of the `std` namespace. That's the
     // first frame in user-code that ended up triggering the
     // verbose_trap. Hence that's the one we want to display.
-    if (!frame_name.GetStringRef().starts_with("std::"))
+    //
+    // IsHidden will get us to the first non-implementation detail
+    // frame. But that could still be in the `std` namespace, so
+    // check the namespace prefix too.
+    if (!frame_name.GetStringRef().starts_with("std::") &&
+        !most_relevant_frame_sp->IsHidden()
+#ifdef LLDB_ENABLE_SWIFT
+        // In Swift-C++ interop, we generate frames with a "std."
+        // prefix for functions from libc++. We don't want to
+        // stop in those frames either.
+        && !frame_name.GetStringRef().starts_with("__C.std.") &&
+        !frame_name.GetStringRef().starts_with("std.")
```

Comment:
shouldn't this check live inside SwiftHiddenFrameRecognizer(), assuming that we never want to stop there?

Reply:
Depends, do we want to hide *all* frames from the `std` module? With C++ interop that could make sense. But wasn't sure.